tweet_recommendations/data_processing/udi_processor.py

```diff
@@ -0,0 +1,197 @@
+import argparse
+import os
+import sys
+from collections import deque
+from functools import partial
+from multiprocessing import Pool
+from pathlib import Path
+from typing import Dict, List, Tuple, Union
+
+import pandas as pd
+from tqdm import tqdm
+
+TWEET_SEPARATOR = "***"
+VALUE_SEPARATOR = ":"
+MULTILINE_FIELDS = ["Text", "Origin"]
+SINGLE_FIELDS = [
+    "Type",
+    "URL",
+    "ID",
+    "Time",
+    "RetCount",
+    "Favorite",
+    "MentionedEntities",
+    "Hashtags",
+]
+
+
+def udi_parse_dataset(
+    dataset_dir: Union[str, Path],
+    output_dir: Union[str, Path],
+    verbose: bool = False,
+    minimal_hashtags: int = 0,
+    parallelism: int = 2,
+) -> None:
+    """Parse whole dataset in UDI format, save results as structured DataFrames
```

Comment:
Jaka to jest konwencja komentowania? Ja i Kuba używaliśmy reStructuredText, przydałoby się to ujednolicić w którąś ze stron

Reply:
Racja, ja tutaj z przyzwyczajenia użyłem konwencji docstringów z [PEP 257](https://www.python.org/dev/peps/pep-0257/), ale możemy zmienić na reST.

Reply:
W sumie nie mam zdania, reST wydaje mi się schludniejsze, ale warto się trzymać, skoro wszystkie inne pliki są już w tej formie